src/lambda/commands/downloadLambda.ts

```diff
@@ -232,3 +235,57 @@ function computeLambdaRoot(lambdaLocation: string, functionNode: LambdaFunctionN
 
     return lambdaIndex > -1 ? normalizedLocation.slice(0, lambdaIndex) : path.dirname(normalizedLocation)
 }
+
+export type ApplicationJson = {
+    ApplicationName?: string
+    DeploymentMethod: 'lambda' | 'cloudformation'
+    Functions: {
+        [logicalId: string]: {
+            PhysicalId: {
+                [region: string]: string
+            }
+        }
+    }
+    StackName?: string
+}
+
+/**
+ * Create and write a '.application.json' file to hold metadata about
+ * the downloaded lambda function
+ * @param lambdaFunc Lambda function
+ * @param region aws region
+ * @param downloadLocation dir path where the file will be written
+ */
+export function createApplicationJson(
+    lambdaFunc: Lambda.GetFunctionResponse,
+    region: string,
+    downloadLocation: string
+) {
+    const physicalId = lambdaFunc.Configuration?.FunctionName ?? ''
+    const logicalId = lambdaFunc.Tags?.['aws:cloudformation:logical-id'] ?? physicalId
+    const stackName = lambdaFunc.Tags?.['aws:cloudformation:stack-name'] ?? undefined
+    const deploymentMethod = stackName !== undefined ? 'cloudformation' : 'lambda'
+    const appJson: ApplicationJson = {
+        DeploymentMethod: deploymentMethod,
+        Functions: {
+            [logicalId]: {
+                PhysicalId: {
+                    [region]: physicalId,
+                },
+            },
+        },
+        StackName: stackName,
+    }
+
+    writeAppJsonFile(downloadLocation, appJson)
+}
+
+function writeAppJsonFile(destinationPath: string, appJson: ApplicationJson) {
+    // Should not block download lambda if there is an error writing .application.json file
+    try {
+        fs.writeFileSync(path.join(destinationPath, '.application.json'), JSON.stringify(appJson))
```

Comment:
because `.application.json` is a file that customers will potentially want to inspect, and because the data is tiny, let's make the json human-readable:

```suggestion
        fs.writeFileSync(path.join(destinationPath, '.application.json'), JSON.stringify(appJson, undefined, 4))
```